modules/apps/transfer/keeper/relay.go

```diff
@@ -326,48 +344,52 @@ func (k Keeper) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Pac
 
 // OnTimeoutPacket refunds the sender since the original packet sent was
 // never received and has been timed out.
-func (k Keeper) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Packet, data types.FungibleTokenPacketData) error {
+func (k Keeper) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Packet, data types.FungibleTokenPacketDataV2) error {
 	return k.refundPacketToken(ctx, packet, data)
 }
 
 // refundPacketToken will unescrow and send back the tokens back to sender
 // if the sending chain was the source chain. Otherwise, the sent tokens
 // were burnt in the original send so new tokens are minted and sent to
 // the sending address.
-func (k Keeper) refundPacketToken(ctx sdk.Context, packet channeltypes.Packet, data types.FungibleTokenPacketData) error {
+func (k Keeper) refundPacketToken(ctx sdk.Context, packet channeltypes.Packet, data types.FungibleTokenPacketDataV2) error {
 	// NOTE: packet data type already checked in handler.go
 
-	// parse the denomination from the full denom path
-	trace := types.ParseDenomTrace(data.Denom)
+	for _, token := range data.Tokens {
+		fullDenomPath := token.GetFullDenomPath()
 
-	// parse the transfer amount
-	transferAmount, ok := sdkmath.NewIntFromString(data.Amount)
-	if !ok {
-		return errorsmod.Wrapf(types.ErrInvalidAmount, "unable to parse transfer amount (%s) into math.Int", data.Amount)
-	}
-	token := sdk.NewCoin(trace.IBCDenom(), transferAmount)
+		// parse the denomination from the full denom path
+		trace := types.ParseDenomTrace(fullDenomPath)
 
-	// decode the sender address
-	sender, err := sdk.AccAddressFromBech32(data.Sender)
-	if err != nil {
-		return err
-	}
+		// parse the transfer amount
+		transferAmount, ok := sdkmath.NewIntFromString(token.Amount)
+		if !ok {
+			return errorsmod.Wrapf(types.ErrInvalidAmount, "unable to parse transfer amount (%s) into math.Int", transferAmount)
+		}
+		token := sdk.NewCoin(trace.IBCDenom(), transferAmount)
```

Comment:
maybe lets rename to `coin` given that there's another variable here also named `token` with a different type and denom value :sweat_smile: